pkg/network/dhcp/masquerade_test.go

```diff
@@ -58,22 +58,29 @@ var _ = Describe("Masquerade DHCP configurator", func() {
 		var vmiSpecNetwork *v1.Network
 		var vmiSpecIface *v1.Interface
 
-		generateExpectedConfig := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
+		generateExpectedConfigIPv6Disabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
 			ipv4Gateway, ipv4, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv4)
-			ipv6Gateway, ipv6, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv6)
 			expectedConfig := cache.DHCPConfig{Name: ifaceName,
-				IP:                  *ipv4,
-				IPv6:                *ipv6,
-				Mtu:                 uint16(mtu),
-				AdvertisingIPAddr:   ipv4Gateway.IP.To4(),
-				AdvertisingIPv6Addr: ipv6Gateway.IP.To16(),
-				Gateway:             ipv4Gateway.IP.To4(),
+				IP:                *ipv4,
+				Mtu:               uint16(mtu),
+				AdvertisingIPAddr: ipv4Gateway.IP.To4(),
+				Gateway:           ipv4Gateway.IP.To4(),
 			}
 
 			if macString != nil {
 				mac, _ := net.ParseMAC(*macString)
 				expectedConfig.MAC = mac
 			}
+
+			return expectedConfig
+		}
+
+		generateExpectedConfigIPv6Enabled := func(vmiSpecNetwork *v1.Network, macString *string, mtu int, ifaceName string) cache.DHCPConfig {
+			expectedConfig := generateExpectedConfigIPv6Disabled(vmiSpecNetwork, macString, mtu, ifaceName)
+			ipv6Gateway, ipv6, _ := virtnetlink.GenerateMasqueradeGatewayAndVmIPAddrs(vmiSpecNetwork, iptables.ProtocolIPv6)
```

Comment:
why you don't check err ? (3rd arg)

Reply:
I don't know. It was like this in the original function: 
https://github.com/kubevirt/kubevirt/blob/main/pkg/network/dhcp/masquerade_test.go#L62-L63

Reply:
maybe worth to check on a following PR, whatever you think

Reply:
This specific line was not added. But yes I agree this can be in a followup.

Reply:
Addressed in the second commit 
https://github.com/kubevirt/kubevirt/pull/6516/commits/9b371b17d48e9dd7b4299ae0fe45d3a239cb7da0